### ISSUE TITLE
SaveChanges after BulkUpdate

### ISSUE THREAD

**Issue by mdnaseri** [zzzprojects/EntityFramework-Extensions#142]
Hello,
I am using a BulkUpdate for updating 7000 records.
the BulkUpdate works great and all data updated but when I update another one row after that and using savechanges() to store data, sql profiler shows that all of 7000 records updated again.
How can I handle it?

code:

var memprice = en.Table_MemPrice.Where(oo => oo.f_ActionID == Deposit.f_DepositID && MemberGUID.Contains(oo.f_MemberGUID)).ToList();
memprice.ForEach(x => { x.f_Active = true; x.f_Cal= true; x.f_AddDate = SysDate;x.f_AddUserID = UserID; });

en.BulkUpdate(memprice, options => options.ColumnInputExpression = c => new { c.f_PriceMemberID,c.f_Active, c.f_Cal,c.f_AddDate,c.f_AddUserID});

var dep =en.Table_Dep.Where(oo => oo.f_DepositID == Deposit.f_DepositID).First();
dep.f_DepositStatusID = 4;
en.SaveChanges();

**Comment by JonathanMagnan** [comment]
Hello @mdnaseri ,

My bad, I forget that DetectChanges should be disabled before modifying the state.

I have added a task to look on our side instead and make sure it doesn't take 15min ;)

Best Regards,

Jonathan

**Comment by JonathanMagnan** [comment]
Hello @mdnaseri ,

We have made some improvement in this request. It's now possible on our side to attach/change state of multiple entities without impacting the performance (A little bit such as `Add` vs `AddRange` performance).

We are currently looking to add some `chaining` operation as we did in [Dapper Plus](http://dapper-plus.net/). So we will let people soon to change state and/or even chain some bulk operations.

It might take a little bit more time than we expected since we want to add some functionalities with this change.

If you want, we could also provide you temporary an extension method to update on your side the state without impacting the performance.

Best Regards,

Jonathan
